my-app/src/app/core/service/authentication.service.ts

```diff
@@ -15,22 +15,38 @@ export class AuthenticationService {
 
   constructor(private httpService: HttpService) {}
 
-  login(value: UserLogin): Observable<UserInfo> {
+  public login(value: UserLogin): Observable<UserInfo> {
     return this.httpService.getAuthToken(value).pipe(
       switchMap((token) => this.httpService.getUserInfo(token)),
-      tap((value)=> window.localStorage.setItem(this.userAuthKey, JSON.stringify(value)))
+      tap((value)=> {
+        return window.localStorage.setItem(this.userAuthKey, JSON.stringify(value))
+      })
     );
   }
 
-  logout(): void {
+  public logout(): void {
     window.localStorage.removeItem(this.userAuthKey);
   }
 
-  isUserAuthenticated(): boolean {
+  public isUserAuthenticated(): boolean {
     return this.getUserInfo() !== null;
   }
 
-  getUserInfo(): UserInfo {
+  public getUserInfo(): UserInfo {
     return JSON.parse(window.localStorage.getItem(this.userAuthKey));
   }
+
+  public getUserData(): Observable<UserInfo> {
+    return new Observable((obs) => {
```

Comment:
you don't need to create a new observable here. Use build-in operators instead. Think of another places where you can use build-in operators
` return of(userInfo)`